rollup.config.mjs

```diff
@@ -16,14 +18,20 @@ for (const input of inputFiles) {
 	rmSync(input.replace('.mjs', '.cjs'), { force: true });
 }
 
+const rootDir = fileURLToPath(new URL('.', import.meta.url));
+
 export default inputFiles.map((input) => {
 	return {
 		input,
 		output: {
 			format: 'cjs',
-			file: input.replace('.mjs', '.cjs'),
+			dir: rootDir,
+			entryFileNames: ({ facadeModuleId }) => {
+				return relative(rootDir, facadeModuleId).replace('.mjs', '.cjs');
+			},
```

Comment:
[note] When enabling `preserveModules`, `dir` is needed, and then `entryFileNames` is also needed to output a file ending with `.cjs`.